pkg/database/database.go

```diff
@@ -83,13 +86,22 @@ func ConnectDB(
 
 		sql.Register(SQLiteCustomDriverName, &sqlite3.SQLiteDriver{
 			ConnectHook: func(conn *sqlite3.SQLiteConn) error {
+				// create LRU cache to cache regexp statements and results.
+				cache, err := lru.New[string, *regexp.Regexp](1000)
+				if err != nil {
+					return eris.Wrap(err, "error creating lru cache to cache regexp statements")
+				}
 				return conn.RegisterFunc("regexp", func(re, s string) bool {
-					c, err := regexp.Compile(re)
-					if err != nil {
-						return false
+					result, ok := cache.Get(re)
+					if !ok {
+						complied, err := regexp.Compile(re)
```

Comment:
I think you should assign it straight to `result` instead so you avoid the duplicate `return MatchString`

Reply:
done.